scripts/ci/run_perftest.sh

```diff
@@ -117,36 +151,73 @@ for scale in 50 100 500; do
     # run ingest tests
     cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\"}" > $working_dir/test-ingest-$scale_string-batchsize-$batch-workers-$workers.json
 
-    bucket_id=$(curl -H "Authorization: Token $TEST_TOKEN" "http://${NGINX_HOST}:8086/api/v2/buckets?org=$TEST_ORG" | jq -r ".buckets[] | select(.name | contains(\"$db_name\")).id")
-    sleep 1
-    # stop daemon and force compaction
-    systemctl stop influxdb
-    set +e
-    shards=$(find /var/lib/influxdb/engine/data/$bucket_id/autogen/ -maxdepth 1 -mindepth 1)
-    set -e
-    for shard in $shards; do
-      /home/ubuntu/influx_tools compact-shard -force -verbose -path $shard
-    done
-
-    # restart daemon
-    systemctl unmask influxdb.service
-    systemctl start influxdb
+    force_compaction
 
     # run influxql and flux query tests
     for query_file in $query_files; do
       format=$(echo $query_file | cut -d '-' -f7)
       cat ${DATASET_DIR}/$query_file | ${GOPATH}/bin/query_benchmarker_influxdb --urls=http://localhost:8086 --debug=0 --print-interval=0 --workers=$workers --json=true --organization=$TEST_ORG --token=$TEST_TOKEN | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\", query_format: \"$format\"}" > $working_dir/test-query-$scale_string-format-$format-workers-$workers.json
     done
 
-    # delete db to start anew
-    delete_response=$(curl -s -o /dev/null -X DELETE -H "Authorization: Token $TEST_TOKEN" "http://${NGINX_HOST}:8086/api/v2/buckets/$bucket_id?org=$TEST_ORG" -w %{http_code})
-    if [ $delete_response != "204" ]; then
-      echo "bucket not deleted!"
-      exit 1
-    fi
+    clean_db
   done
 done
 
+# Metaquery tests
+
+# Controls how many data points will be generated. The total number of points
+# will be scale_var * scale_var, where each point is (approximately) a unique
+# series.
+scale_var=1000
+
+# How many queries to generate.
+queries=100
+
+# The time range controls both the span over which data is generated, and the
+# span over which queries will be performed. timestamp-start and timestamp-end
+# must be provided to both the data generation and query generation commands
+# and must be the same to ensure that the queries cover the data range.
+start="2019-01-01T00:00:00Z"
+end="2020-01-01T00:00:00Z"
+
+# How long to run the query phase of the benchmarks. It's best to specify a
+# duration to limit the maximum amount of time the queries can run for these,
+# since individual queries can take a long time to run.
+duration=30s
+
+# Generate data
+data_fname="influx-bulk-records-usecase-metaquery.txt"
```

Comment:
Will there ever be a case where you want to run this test against different values of the scale_var? if so, you might append $scale_var here to the filename so that you can differentiate later on.

Reply:
Also it may help to include the seed value in the filename so that the dataset can be reproduced as faithfully as possible in the event of a deep debugging / reproducer session.